sopht/utils/restart_sim.py

```diff
@@ -0,0 +1,46 @@
+from typing import Type, Union
+import os
+import elastica as ea
+from sopht.simulator.flow.navier_stokes_flow_simulators import (
+    UnboundedNavierStokesFlowSimulator3D,
+    UnboundedNavierStokesFlowSimulator2D,
+)
+from sopht.simulator.flow.passive_transport_flow_simulators import (
+    PassiveTransportFlowSimulator,
+)
+from sopht.utils.io import IO
+
+
+def restart_simulation(
+    flow_sim: Union[
+        UnboundedNavierStokesFlowSimulator2D,
+        UnboundedNavierStokesFlowSimulator3D,
+        PassiveTransportFlowSimulator,
+    ],
+    restart_simulator: Type[ea.BaseSystemCollection],
+    io: IO,
+    rod_io: IO,
+    forcing_io: IO,
+    restart_dir: str,
+) -> None:
+    # find latest saved data
+    iter_num = []
+    for filename in os.listdir():
+        if "sopht" in filename and "h5" in filename:
+            iter_num.append(int(filename.split("_")[-1].split(".")[0]))
+
+    if len(iter_num) == 0:
+        print("There is no file to load in the directory")
+        return
+
+    latest = max(iter_num)
+    # load sopht data
+    flow_sim.time = io.load(h5_file_name=f"sopht_{latest:04d}.h5")
+    _ = rod_io.load(h5_file_name=f"rod_{latest:04d}.h5")
```

Comment:
Do we need the placeholders here?

Reply:
I think we do. To load the correct file (latest saved file).

Reply:
No I mean the `_ = `. I don't think it's here since we don't need the return value.
```suggestion
    rod_io.load(h5_file_name=f"rod_{latest:04d}.h5")
```